packages/core/addon/serializers/dashboard.js

```diff
@@ -22,6 +22,16 @@ export default AssetSerializer.extend({
     return this._super(...arguments);
   },
 
+  serialize(snapshot) {
+    const filterSources = snapshot.attr('filters').map(filter => filter.attr('dimension').source);
+    const dashboard = this._super(...arguments);
+    dashboard.data.attributes.filters = dashboard.data.attributes.filters.map((filter, i) => {
+      filter.dimension = filterSources[i] ? `${filterSources[i]}.${filter.dimension}` : filter.dimension;
+      return filter;
+    });
+    return dashboard;
```

Comment:
```suggestion
    const filterSources = snapshot.attr('filters').reduce((dimensionSources, filter) => { 
      const { name, source } = filter.attr('dimension');
      dimensionSources[name] = [...(dimensionSources[name] || []), source]; 
      return dimensionSources
    }, {});
    const dashboard = this._super(...arguments);
    dashboard.data.attributes.filters = dashboard.data.attributes.filters.map((filter, i) => {
      const { dimension } = filter;
      filter.dimension = filterSources[dimension] ? `${filterSources[dimension]}.${dimension}` : dimension;
      return filter;
    });
    return dashboard;
```
Could we do something like this so we're not relying on the order of each list before and after calling super? I think it might be more robust if we use the dimension id as the key for assigning the source in the filters.

Reply:
Problem is that we need to account for different datasources that have the same dimension name. Because we might have dimensions with the same base name but different valid values to filter by.

Best I could do is build a key with the dimension name AND values.